nixos/modules/services/matrix/conduwuit.nix

```
@@ -0,0 +1,245 @@
+              listening on an address will be disabled. The `address` option must be set to
+              `null` (the default value). You must add your reverse proxy to the `conduwuit`
+              group, unless RW permissions are specified with `unix_socket_perms` (666 minimum).
+            '';
+          };
+          global.unix_socket_perms = lib.mkOption {
+            type = lib.types.ints.positive;
+            default = 660;
+            description = "The default permissions (in octal) to create the UNIX socket with.";
+          };
+          global.max_request_size = lib.mkOption {
+            type = lib.types.ints.positive;
+            default = 20000000;
+            description = "Max request size in bytes. Don't forget to also change it in the proxy.";
+          };
+          global.allow_registration = lib.mkOption {
+            type = lib.types.bool;
+            default = false;
+            description = ''
+              Whether new users can register on this server.
+
+              If set to true without a token configured, users can register with no
+              form of 2nd-step only if you set
+              `yes_i_am_very_very_sure_i_want_an_open_registration_server_prone_to_abuse` to
+              true in your config.
+              If you would like registration only via token reg, please configure
+              `registration_token` or `registration_token_file`.
+            '';
+          };
+          global.allow_encryption = lib.mkOption {
+            type = lib.types.bool;
+            default = true;
+            description = "Whether new encrypted rooms can be created. Note: existing rooms will continue to work.";
+          };
+          global.allow_federation = lib.mkOption {
+            type = lib.types.bool;
+            default = true;
+            description = ''
+              Whether this server federates with other servers.
+            '';
+          };
+          global.trusted_servers = lib.mkOption {
+            type = lib.types.listOf lib.types.str;
+            default = [ "matrix.org" ];
+            description = ''
+              Servers listed here will be used to gather public keys of other servers
+              (notary trusted key servers).
+
+              Currently, conduwuit doesn't support inbound batched key requests, so
+              this list should only contain other Synapse servers.
+
+              Example: `[ "matrix.org" "constellatory.net" "tchncs.de" ]`
+            '';
+          };
+          global.database_path = lib.mkOption {
+            type = lib.types.str;
+            default = "/var/lib/conduwuit/";
+            readOnly = true;
+            description = ''
+              Path to the conduwuit database, the directory where conduwuit will save its data.
+              Note that due to using the DynamicUser feature of systemd, this value should not be changed
+              and is set to be read only.
```

```suggestion
            description = ''
              Path to the conduwuit database, the directory where conduwuit will save its data.
              To change this value the DynamicUser feature has to be disabled for the conduwuit
              systemd service.
```
Move this to a assertion since there is no way to override a read only option.